src/components/Property/WorkOrdersHistory/WorkOrdersHistoryTable.js

```diff
@@ -37,44 +39,117 @@ const WorkOrdersHistoryTable = ({
-  )
+  const clearFilters = () => {
+    setTradeToFilterBy(null)
+  }
+
+  if (!tradeToFilterBy && workOrders?.length > 0) {
+    return (
+      <>
+        <h2 className="lbh-heading-h2">{tabName}</h2>
+        <WorkOrdersHistoryFilter
+          handleChange={handleChange}
+          clearFilters={clearFilters}
+        />
+        <Table className="govuk-!-margin-top-5 work-orders-history-table">
+          <THead>
+            <TR className="lbh-body">
+              {user && canAccessWorkOrder(user) && (
+                <TH scope="col">Reference</TH>
+              )}
+              <TH scope="col">Date raised</TH>
+              <TH scope="col">Trade</TH>
+              <TH scope="col">Status</TH>
+              <TH scope="col">Description</TH>
+            </TR>
+          </THead>
+          <TBody>
+            {workOrders.map((workOrder, index) => (
+              <WorkOrdersHistoryRow
+                key={index}
+                reference={workOrder.reference}
+                dateRaised={new Date(workOrder.dateRaised)}
+                tradeDescription={workOrder.tradeDescription}
+                description={workOrder.description}
+                status={workOrder.status}
+              />
+            ))}
+          </TBody>
+        </Table>
+        {workOrders && renderLoadMoreWorkOrders()}
+      </>
+    )
+  }
+
+  if (filteredOrders?.length > 0) {
+    return (
+      <>
+        <h2 className="lbh-heading-h2">{tabName}</h2>
+        <WorkOrdersHistoryFilter
+          handleChange={handleChange}
+          clearFilters={clearFilters}
+        />
+        <Table className="govuk-!-margin-top-5 work-orders-history-table">
+          <THead>
+            <TR className="lbh-body">
+              {user && canAccessWorkOrder(user) && (
+                <TH scope="col">Reference</TH>
+              )}
+              <TH scope="col">Date raised</TH>
+              <TH scope="col">Trade</TH>
+              <TH scope="col">Status</TH>
+              <TH scope="col">Description</TH>
+            </TR>
+          </THead>
+          <TBody>
+            {filteredOrders.map((workOrder, index) => (
+              <WorkOrdersHistoryRow
+                key={index}
+                reference={workOrder.reference}
+                dateRaised={new Date(workOrder.dateRaised)}
+                tradeDescription={workOrder.tradeDescription}
+                description={workOrder.description}
+                status={workOrder.status}
+              />
+            ))}
+          </TBody>
+        </Table>
+      </>
+    )
+  }
+
+  if (tradeToFilterBy && filteredOrders.length === 0) {
+    return (
+      <>
+        <h2 className="lbh-heading-h2">{tabName}</h2>
+        <WorkOrdersHistoryFilter
+          handleChange={handleChange}
+          clearFilters={clearFilters}
+        />
+        <h4 className="lbh-heading-h4">
+          There are no historical repairs with {tradeToFilterBy}.
+        </h4>
+      </>
+    )
+  }
 }
```

Comment:
There is a lot of repeated logic here. For a start, the following HTML is always used 
```jsx
  <>
        <h2 className="lbh-heading-h2">{tabName}</h2>
        <WorkOrdersHistoryFilter
          handleChange={handleChange}
          clearFilters={clearFilters}
        />
       ///
      </>
```

So maybe find a way to always return that?

Reply:
Secondly, the only difference between the two tables is the list used. Instead, you could add a condition for which list is used.

![image](https://github.com/user-attachments/assets/7265f8e5-dc7f-4fb9-bdb2-6ada7447aef2)